byteparse/shared/src/main/scala/byteparse/classparse/CodeParser.scala

```diff
@@ -0,0 +1,514 @@
+  case class IfNe(offset: Short) extends OpCode
+  case class IfNonNull(offset: Short) extends OpCode
+  case class IfNull(offset: Short) extends OpCode
+
+  case class ImpDep1() extends OpCode
+  case class ImpDep2() extends OpCode
+
+  case class InstanceOf(index: Int) extends OpCode
+  case class InvokeDynamic(index: Int) extends OpCode
+  case class InvokeInterface(index: Int, count: Int) extends OpCode
+  case class InvokeSpecial(index: Int) extends OpCode
+  case class InvokeStatic(index: Int) extends OpCode
+  case class InvokeVirtual(index: Int) extends OpCode
+
+  case class JSR(offset: Short) extends OpCode
+  case class JSRW(offset: Int) extends OpCode
+
+  case class L2D() extends OpCode
+  case class L2F() extends OpCode
+  case class L2I() extends OpCode
+  case class LAdd() extends OpCode
+  case class LSub() extends OpCode
+  case class LMul() extends OpCode
+  case class LNeg() extends OpCode
+  case class LDiv() extends OpCode
+  case class LRem() extends OpCode
+
+  case class LAnd() extends OpCode
+  case class LOr() extends OpCode
+  case class LShl() extends OpCode
+  case class LShr() extends OpCode
+  case class LUShr() extends OpCode
+  case class LXor() extends OpCode
+
+  case class LALoad() extends OpCode
+  case class LAStore() extends OpCode
+  case class LCmp() extends OpCode
+  case class LConst0() extends OpCode
+  case class LConst1() extends OpCode
+  case class LLoad(index: Int) extends OpCode
+  case class LLoad0() extends OpCode
+  case class LLoad1() extends OpCode
+  case class LLoad2() extends OpCode
+  case class LLoad3() extends OpCode
+  case class LStore(index: Int) extends OpCode
+  case class LStore0() extends OpCode
+  case class LStore1() extends OpCode
+  case class LStore2() extends OpCode
+  case class LStore3() extends OpCode
+  case class LReturn() extends OpCode
+
+  case class LDC(index: Int) extends OpCode
+  case class LDCW(index: Int) extends OpCode
+  case class LDC2W(index: Int) extends OpCode
+
+  case class LookUpSwitch(defaultOffset: Int, pairs: Seq[(Int, Int)]) extends OpCode
+
+  case class MonitorEnter() extends OpCode
+  case class MonitorExit() extends OpCode
+
+  case class MutliANewArray(index: Int, dimensions: Int) extends OpCode
+  case class New(index: Int) extends OpCode
+  case class NewArray(atype: Byte) extends OpCode
+
+  case class Nop() extends OpCode
+  case class Pop() extends OpCode
+  case class Pop2() extends OpCode
+  case class Swap() extends OpCode
+  case class PutField(index: Int) extends OpCode
+  case class PutStatic(index: Int) extends OpCode
+  case class Ret(index: Int) extends OpCode
+  case class Return() extends OpCode
+
+  case class SALoad() extends OpCode
+  case class SAStore() extends OpCode
+  case class SIPush(short: Short) extends OpCode
+
+  case class TableSwitch(defaultOffset: Int, low: Int, high: Int, offsets: Seq[Int]) extends OpCode
+
+  case class Wide() extends OpCode // correct behavior for this command hasn't been implemented yet
```

Comment:
I think we need a really huge java file, where size of addresses in the code go beyond 1 or 2 bytes. And since `ProjectTests` passes correctly, it almost never happens as I think.